src/richie/apps/search/filter_definitions/courses.py

```diff
@@ -143,16 +225,55 @@ def get_faceted_definitions(self, facets, *args, **kwargs):
         #       }
         #   }
         # 👆 becomes 👇
-        #   {"A": x, "B": y, "C": z}
-        key_count_map = reduce(
-            lambda agg, key_count_dict: {
-                **agg,
-                key_count_dict["key"]: key_count_dict["doc_count"],
-            },
-            facets[self.name][self.name]["buckets"],
-            {},
+        #   [
+        #       ("A", x),
+        #       ("B", y),
+        #       ("C", z),
+        #   ]
+        # sorted from highest facet count to lowest.
+        key_count_pairs = sorted(
+            [
+                (key_count_dict["key"], key_count_dict["doc_count"])
+                for key_count_dict in facets[self.name][self.name]["buckets"]
+            ],
+            key=itemgetter(1),
+            reverse=True,
         )
 
+        # We requested more facets than needed to be able to determine with certainty if there
+        # are more available than what we're returning (see in `get_aggs_fragment`).
+        base_facet_limit = applicable_facet_limit(data, self.name)
+        control_length = len(data[self.name]) + 1
+        # We have as many results as we requested: there are more values (at least the `+1` we
+        # added to our request besides the base limit and the active values)
+        if len(key_count_pairs) == base_facet_limit + control_length:
+            has_more_values = True
+        # We have no value(s) besides the base limit: we can safely conclude there are no more
+        # values (eg. we did not get that requested `+1`)
+        elif len(key_count_pairs) <= base_facet_limit:
+            has_more_values = False
+        # We are somewhere between one more than the base limit and the requested number: this is
+        # ambiguous because it depends on the placement of our active values (which are always
+        # included no matter the circumstances).
+        # If at least one of the values beyond base limit is not an active value, we know we're not
+        # going to show it. Therefore there are more values.
+        elif [
+            value
+            for value in [key_count_pair[0] for key_count_pair in key_count_pairs][
```

Comment:
I think one check can do it all:
```python
has_more_values = any(filter(lambda c: c[0] not in data[self.name], key_counts_pairs[base_facet_limit:]))
```
Because:
```
if len(key_count_pairs) == base_facet_limit + control_length:
    then there is necessarily 1 of the extra values that is not a selected facet (the +1) so it is True

if len(key_count_pairs) <= base_facet_limit:
    then key_counts_pairs[base_facet_limit:] is equal to [] and it is False
```